source/fundamentals/connection/mongoclientsettings.txt

```diff
@@ -0,0 +1,197 @@
+
+For a list of methods available in ``MongoClientSettings``, 
+see the `API Documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html>`__.
+
+The following sections describe the commonly used ``MongoClientSettings``:
+
+- :ref:`Cluster Settings <mcs-cluster-settings>`
+- :ref:`Socket Settings <mcs-socket-settings>`
+- :ref:`Connection Pool Settings <mcs-connectionpool-settings>`
+- :ref:`Server Settings <mcs-server-settings>`
+- :ref:`SSL Settings <mcs-ssl-settings>`
+
+.. _mcs-cluster-settings:
+
+Cluster Settings
+----------------
+
+Use the `applyToClusterSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToClusterSettings(com.mongodb.Block)>`__
+method to specify settings on how to connect to a MongoDB cluster. 
+
+Example
+~~~~~~~
+
+This example specifies the following settings for connecting to a
+MongoDB cluster:
+
+- To use the settings from your ``ConnectionString``
+- To connect to ``MULTIPLE`` servers in your cluster
+- The server selection to take at most ``5 SECONDS``
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin ClusterSettings
+   :end-before: end ClusterSettings
+   :language: java
+   :emphasize-lines: 3-6
+   :dedent:
+
+For a list of methods you can use to specify your cluster settings, see
+the `API Documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/ClusterSettings.Builder.html>`__.
+
+.. _mcs-socket-settings:
+
+Socket Settings
+---------------
+
+Use the `applyToSocketSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToSocketSettings(com.mongodb.Block)>`__
+method to specify settings on how to connect to a MongoDB server.
+
+Example
+~~~~~~~
+
+This example specifies the following settings for connecting to a
+MongoDB socket:
+
+- The connect to the socket within ``1000 MILLISECONDS``
+- To read from the socket within ``10 SECONDS``
+
+.. literalinclude:: /includes/fundamentals/code-snippets/mcs.java
+   :start-after: begin SocketSettings
+   :end-before: end SocketSettings
+   :language: java
+   :emphasize-lines: 4-5
+   :dedent:
+
+For a list of methods you can use to specify your socket settings, see
+the `API Documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/connection/SocketSettings.Builder.html>`__.
+
+.. _mcs-connectionpool-settings:
+
+Connection Pool Settings
+------------------------
+
+Use the `applyToConnectionPoolSettings() <{+api+}/apidocs/mongodb-driver-core/com/mongodb/MongoClientSettings.Builder.html#applyToConnectionPoolSettings(com.mongodb.Block)>`__
+method to specify settings related to a pool of connections to a MongoDB server.
+
+Example
+~~~~~~~
+
+This example specifies the following settings for connecting to a
+MongoDB connection pool:
+
+- The thread to wait at most ``5 SECONDS`` for an available connection
+- To have at most one connection associated with the pool
```

Comment:
issue: This wouldn't be a pool, and I am strongly opposed to including this value. A user could copy this and then would experience very poor performance (if not an outright seized application).

Better would be to discuss the _implications_ of these options. Why do we support connection pooling? What are some drawbacks with higher, and lower, values?